packages/dropzone/stories/dropzone.stories.ts

```diff
@@ -44,48 +50,140 @@ type StoryArgs = {
-                    </label>
-                    <input type="file" id="file-input" style="display: none" />
-                </div>
-                <div slot="description">
-                    or
-                    <sp-link href="http://stock.adobe.com" target="blank">
-                        Search Adobe Stock
-                    </sp-link>
-                </div>
             </sp-illustrated-message>
+            <div>
+                <label for="file-input">
+                    <sp-link
+                        href="javascript:;"
+                        onclick="this.parentElement.nextElementSibling.click()"
+                    >
+                        Select a File
+                    </sp-link>
+                    from your computer
+                </label>
+                <input type="file" id="file-input" style="display: none" />
+            </div>
+            <div>
+                or
+                <sp-link href="http://stock.adobe.com" target="blank">
+                    Search Adobe Stock
+                </sp-link>
+            </div>
         </sp-dropzone>
     `;
 };
 Dragged.args = {
     isDragged: true,
 };
+
+class ControlledDropzone extends LitElement {
+    private fileName = 'mock_file.png';
+
+    @state()
+    private input?: string = undefined;
+
+    override render(): TemplateResult {
+        return html`
+            <span>
+                ${this.renderDroppedContent()}
+                <sp-dropzone
+                    tabindex="0"
+                    id="dropzone"
+                    drop-effect="copy"
+                    ?dragged=${this.input !== undefined}
+                    @sp-dropzone-drop=${this.onChange}
+                >
+                    <sp-illustrated-message heading="Drag and Drop Your File">
+                        ${illustration}
+                    </sp-illustrated-message>
+                    <div>
+                        <label for="file-input">
+                            <sp-link
+                                href="javascript:;"
+                                onclick="this.parentElement.nextElementSibling.click()"
+                            >
+                                Select a File
+                            </sp-link>
+                            from your computer
+                        </label>
+                        <input
+                            type="file"
+                            id="file-input"
+                            style="display: none"
+                            @change=${this.onChange}
+                        />
+                    </div>
+                    ${this.renderUpload()}
+                </sp-dropzone>
+            </span>
+        `;
+    }
+
+    private renderDroppedContent(): TemplateResult {
+        return this.input !== undefined
+            ? html`
+                  <sp-action-button style="margin-bottom: 16px;">
+                      Added ${this.fileName}
```

Comment:
I don't think I ever saw this code being run? I might not have been running the demo code correctly though, but `input` was always undefined for me.

Reply:
The button above the dropzone will change in the controlled story after you drag and drop the button into the dropzone or after you "add" a file from your desktop